Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.cpp

```diff
@@ -289,11 +291,14 @@ ObjectPropertyConditionSet generateConditions(JSGlobalObject* globalObject, Stru
             return ObjectPropertyConditionSet::invalid();
         }
 
-        if (!functor(conditions, object)) {
+        if (!functor(conditions, object, structure)) {
             if (ObjectPropertyConditionSetInternal::verbose)
                 dataLog("Functor failed, invalid.\n");
             return ObjectPropertyConditionSet::invalid();
         }
+
+        if (structure != object->structure())
+            return ObjectPropertyConditionSet::invalid();
```

Comment:
Let's not add these branches. They're a bit misleading about this structure not changing. I'd vote for a comment or encoding Structure Object pair in a type.